torch/csrc/autograd/python_function.cpp

```diff
@@ -818,17 +870,33 @@ PyObject * THPFunction_do_backward(THPFunction *self, PyObject *args)
 
 PyObject* THPFunction__register_hook_dict(THPFunction *self, PyObject *_var)
 {
+  HANDLE_TH_ERRORS
   THPUtils_assert(THPVariable_Check(_var), "_register_hook_dict expected a variable");
   THPVariable *var = (THPVariable*)_var;
   std::unique_ptr<FunctionPreHook> hook(new PyFunctionPreHook(
       var->backward_hooks, var->cdata.output_nr()));
-  self->cdata.add_pre_hook(std::move(hook));
+  auto cdata = self->cdata.lock();
+  TORCH_CHECK(cdata,
+    "Legacy autograd function had _register_hook_dict called before the function was "
```

Comment:
This is a very bad error message to tell the user. Please figure out which methods call this one and write something that doesn't mention internal APIs. This gives completely no guidance about the potential fixes.

Reply:
On the other hand it's legacy, so I guess I don't really care.

Reply:
Well, it's easy enough to fix.